### ISSUE TITLE
Container generator fails on names containing spaces.

### ISSUE THREAD

**Issue by nafuti** [apiato/apiato#358]
## Expected Behavior
I would expect the input to be broken up and spaces removed then camel cased the resulting name as required in the different fields.

## Actual Behavior
It throws an error on the commandline especially when generating a Transfomer.

## Steps to Reproduce the Problem
I run the conventional `php artisan apiato:generate:container` command.

## Specifications
  - Apiato Version: latest running code on github.

**Comment by johannesschobel** [comment]
Please do not have ` ` (spaces) in your Container names. There are a lot of functions internally that rely on "non-spaced container names".

Best practice is to have neither camelcase nor other symbols (e.g., `_`)

**Comment by nafuti** [comment]
I gathered as much. But isn't there a way the process can be foolproofed from such behavior happening?

Tried to look for where the input is gotten to see if i can add this check and send a PR but didn't find it.

**Comment by johannesschobel** [comment]
I will do this - i contributed most of the generators so i can easily fix this..

I will "remove all special chars" from the container name and so on.. ;)

**Comment by nafuti** [comment]
About the underscore (_), i think it should allowed and other special characters removed.

**Comment by johannesschobel** [comment]
As the "container name" is used throught Apiato to generate namespaces, i dont think, ` ` (whitespaces) are allowed at all. Honestly, i am not entirely sure about this, but i think, "special characters" (e.g., `_` and so on) aren't allowed either in namespaces! 

Finally, I don't know what the implication of having `_` in the container name are to the framework..

**Comment by nafuti** [comment]
[http://php.net/manual/en/language.variables.basics.php](url)

Thats the naming convention for variables, should probably be cutting across to namespaces.

**Comment by johannesschobel** [comment]
I think, for now, i will go with this for the container-name:
1) replace all ` ` (whitespace) with `` (nothing)
2) replace everything, that is not `a-zA-Z0-9` with `` (nothing)

What do you think?

**Comment by nafuti** [comment]
Ok, i think we can live with that.

**Comment by johannesschobel** [comment]
are you fine with the linked PR?

**Comment by nafuti** [comment]
As long as it gets things done and makes the framework amazing. Thank you.

**Comment by johannesschobel** [comment]
I just tagged the PR as `v2.2.2`. Try pulling the latest changes for `apiato/core` with `composer update`.

**Comment by GabrielNSilva** [comment]
So I can't create a container with composite name using camelCase, like "PointType"?
Because I tried to make it, but I got an error.
The generator created the table name in migration as "pointtypes" but the code is serching for "point_types".
"SQLSTATE[42S02]: Base table or view not found: 1146 Table 'default.point_types' doesn't exist (SQL: select count(*) as aggregate from `point_types` where `point_types`.`deleted_at` is null)"
